src/main/java/com/chrosciu/bootcamp/tasks/github/GithubApplication.java

```diff
@@ -37,8 +45,82 @@ private void dispose() {
 
     @SneakyThrows
     private void run() {
+        // Zadanie 1.a: Flux<Repository> getUserRepositories(String username)
+        // - ma zwracać wszystkie repozytoria użytkownika GH o podanej nazwie
+        testGetUserRepositories();
+
+        // Zadanie 1.b: Flux<Repository> getUsersRepositories(Publisher<String> usernames)
+        // - ma zwracać wszystkie repozytoria podanych użytkowników GH (użytkownicy są przekazani jako strumień)
+        testGetUsersRepositories();
+
+        // Zadanie 1.c: Flux<Branch> getUserRepositoryBranches(String username, String repo)
+        // - ma zwracać nazwy wszystkich branch w repozytorium repo należącym do użytkownika username
+        testGetUserRepositoryBranches();
+
+        // Zadanie 1.d: Flux<String> getAllUserBranches(String username)
+        // - ma zwracać nazwy wszystkich branchy we wszystkich repozytoriach użytkownika GH o podanej nazwie
+        testGetAllUserBranches();
+
+        // Zadanie 4: Użyć metod getAllUserBranches oraz InputUtils.toFlux() w metodzie run() tak aby program “w kółko”:
+        // - pobierał z klawiatury nazwę użytkownika GH
+        // - wyświetlał na konsoli wszystkie repozytoria tego użytkownika
+        testGetUserBranchesFromInputStream();
+
+        // Zadanie 5: Zmienić implementację metody getAllUserBranches tak aby zrównoleglić odpytywanie GH API o nazwy branchy
+        // (zakładamy że kolejność w jakiej te nazwy zostaną zwrócone z metody nie ma znaczenia)
+        testGetAllUserBranchesParallel();
+    }
+
+    private void testGetAllUserBranchesParallel() throws InterruptedException {
+        Flux<String> flux = Flux.just("chrosciu", "baranosiu");
+        CountDownLatch latch = new CountDownLatch(2);
+        flux.subscribe(user -> {
```

Comment:
Zagnieżdżone subscribe to tak średnio wygląda - od tego mamy flatMap żeby się tego pozbyć. 
Ogólnie powinniśmy działać tak za pomocą operatorów poskładać docelowy strumień i dopiero do niego robić subskrypcję.

Reply:
Dzięki.